packages/block-editor/src/components/use-block-drop-zone/index.js

```diff
@@ -358,13 +359,29 @@ export default function useBlockDropZone( {
 					// So, ensure that the drag state is set when the user drags over a drop zone.
 					startDragging();
 				}
+
+				const draggedBlockClientIds = getDraggedBlockClientIds();
+				const targetParents = [
+					targetRootClientId,
+					...getBlockParents( targetRootClientId, true ),
+				];
+
+				// Check if the target is within any of the dragged blocks.
+				const isTargetWithinDraggedBlocks = draggedBlockClientIds.some(
+					( clientId ) => targetParents.includes( clientId )
+				);
+
+				if ( isTargetWithinDraggedBlocks ) {
+					return;
+				}
+
```

Comment:
Can should explain a bit why you had to change the drop zone behavior specifically for this PR?

Reply:
Yes — I removed the `pointer-event:none` for blocks that are being dragged, because it results in drag immediately ending. As a result you can drag the block into the block inner block region. This change prevents inner block regions to be considered as drag targets.

Reply:
See https://github.com/WordPress/gutenberg/pull/67305/files/d4a004941bbcab3431ff14d34a8c3afb4b1cb79e#diff-a95ddb28bdbed1c767b4e7dbaf80d4c8c5af01924f9d4981708fd6f34e8ff210

Reply:
Do you think it would be simpler to enable `useDisabled()` in the props of the block when it starts dragging?

Reply:
What would that give us? We'd still need this code because if you disable any children in the block, you'll still be able to drag things _in between_ the inner blocks.

Reply:
This seems like a robust way to prevent allowing dragging blocks within itself, regardless of HTML or CSS

Reply:
`useDisabled` already exists and disables everything including things that are outside the inner blocks.

Reply:
The goal is to remove this code entirely, if we can't, it's a bad idea.

Reply:
Yeah, we can't, this would the only real fix imo

Reply:
Btw, I'm actually seeing the same issue in trunk, so this fixes an additional issue.

<img width="1352" alt="Screenshot 2024-11-29 at 18 56 47" src="https://github.com/user-attachments/assets/7f1f4940-ddde-459e-bf79-5eaeb9064c95">